auth/auth.go

```diff
@@ -31,40 +51,47 @@ type Resource struct {
 	Name string
 	// Type of resource, e.g.
 	Type string
-}
-
-// Role an account has
-type Role struct {
-	// Name of the role
-	Name string
-	// The resource it has access
-	// TODO: potentially remove
-	Resource *Resource
+	// Endpoint resource e.g NotesService.Create
+	Endpoint string
 }
 
 // Account provided by an auth provider
 type Account struct {
 	// ID of the account (UUIDV4, email or username)
-	Id string `json:"id"`
-	// Token used to authenticate
+	ID string `json:"id"`
+	// Secret used to renew the account
+	Secret *Token `json:"secret"`
+	// Roles associated with the Account
+	Roles []string `json:"roles"`
+	// Any other associated metadata
+	Metadata map[string]string `json:"metadata"`
+}
+
+// Token can be short or long lived
+type Token struct {
+	// The token itself
 	Token string `json:"token"`
-	// Time of Account creation
+	// Type of token, e.g. JWT
+	Type string `json:"type"`
+	// Time of token creation
 	Created time.Time `json:"created"`
-	// Time of Account expiry
+	// Time of token expiry
 	Expiry time.Time `json:"expiry"`
-	// Roles associated with the Account
-	Roles []*Role `json:"roles"`
-	// Any other associated metadata
+	// Subject of the token, e.g. the account ID
+	Subject string `json:"subject"`
+	// Roles granted to the token
+	Roles []string `json:"roles"`
+	// Metadata embedded in the token
 	Metadata map[string]string `json:"metadata"`
 }
 
 const (
-	// MetadataKey is the key used when storing the account
-	// in metadata
+	// MetadataKey is the key used when storing the account in metadata
 	MetadataKey = "auth-account"
-	// CookieName is the name of the cookie which stores the
-	// auth token
-	CookieName = "micro-token"
+	// TokenCookieName is the name of the cookie which stores the auth token
+	TokenCookieName = "micro-token"
+	// SecretCookieName is the name of the cookie which stores the auth secret
+	SecretCookieName = "micro-secret"
```

Comment:
A cookie should not store the secret, this is like storing your password in a cookie. It's going to get passed in every request since cookies are included in headers. I would not feel comfortable having my passwords stored anywhere accept server side.

We may needs refresh tokens, session tokens or to just set UI side to a slightly longer live token.

https://auth0.com/learn/refresh-tokens/
https://auth0.com/blog/refresh-tokens-what-are-they-and-when-to-use-them/

Reply:
I'm prefer to use session and refresh tokens.
Access token has short lifetime is like 1h, and longer refresh.
When refresh token used to issue new access and refresh, it removed from store,

Reply:
Where would the JS client store the secret?